File: demo/freeform-demo.js
// Simulates a script with a free-text prompt no built-in rule can answer —
// exercises the --decider path.
import readline from 'node:readline';

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
const ask = (q) => new Promise((res) => rl.question(q, res));

const name = await ask('Project name: ');
console.log(`Creating project ${JSON.stringify(name)}...`);
console.log('Done.');
rl.close();
process.exit(0);

File: engine-rs/examples/freeform_demo.rs
//! Simulates a script with a free-text prompt no built-in rule can answer —
//! exercises the --decider path.

use std::io::{BufRead, Write};

fn main() {
    print!("Project name: ");
    std::io::stdout().flush().unwrap();
    let mut name = String::new();
    std::io::stdin().lock().read_line(&mut name).unwrap();
    let name = name.trim_end_matches(['\r', '\n']);
    println!("Creating project {name:?}...");
    println!("Done.");
}
